### PR TITLE
feat: [DEL-3256]: apiKey support for ng delegate tokens

### PR DESCRIPTION
**Pull request by arpit3107**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30657)

### CODE REVIEW

File: 120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateTokenNgResource.java [hunk 1]
```diff
@@ -44,13 +46,13 @@
 import javax.ws.rs.QueryParam;
 import lombok.extern.slf4j.Slf4j;
 
-@Api("/ng/delegate-token")
-@Path("/ng/delegate-token")
+@Api("delegate-token-ng")
+@Path("/delegate-token-ng")
```
Comment (abhinav-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateTokenNgResource.java:50]: ng is not req

Comment (abhinav-harness) [reply 120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateTokenNgResource.java:50]: or let it be

File: 400-rest/src/main/java/io/harness/delegate/resources/DelegateNgTokenInternalResource.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  @POST
+  @Timed
+  @ExceptionMetered
+  @InternalApi
+  public RestResponse<DelegateTokenDetails> createToken(
+      @Parameter(description = NGCommonEntityConstants.ACCOUNT_PARAM_MESSAGE) @QueryParam(
+          NGCommonEntityConstants.ACCOUNT_KEY) @NotNull String accountIdentifier,
+      @Parameter(description = NGCommonEntityConstants.ORG_PARAM_MESSAGE) @QueryParam(
+          NGCommonEntityConstants.ORG_KEY) String orgIdentifier,
+      @Parameter(description = NGCommonEntityConstants.PROJECT_PARAM_MESSAGE) @QueryParam(
```
Comment (gaurav-jain1) [inline 400-rest/src/main/java/io/harness/delegate/resources/DelegateNgTokenInternalResource.java:66]: Are orgId and projectId mandatory param for ng api ? If yes, than we can make them @NotNull

File: 400-rest/src/main/java/io/harness/delegate/resources/DelegateNgTokenInternalResource.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+          NGCommonEntityConstants.PROJECT_KEY) String projectIdentifier,
+      @Parameter(description = "Status of Delegate Token (ACTIVE or REVOKED). "
+              + "If left empty both active and revoked tokens will be retrieved") @QueryParam("status")
+      DelegateTokenStatus status) {
+    DelegateEntityOwner owner = DelegateEntityOwnerHelper.buildOwner(orgIdentifier, projectIdentifier);
```
Comment (gaurav-jain1) [inline 400-rest/src/main/java/io/harness/delegate/resources/DelegateNgTokenInternalResource.java:103]: If org and projectIds are optional, than will owner creation throw exception ?

**Review by Vbhatia20 — COMMENTED** [review]
> Reviewed 4 of 6 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30657)* status: 0 of 1 approvals obtained.

**Comment by jennyjames21** [comment]
___
*[400-rest/src/main/java/io/harness/delegate/resources/DelegateNgTokenResource.java, line 80 at r2](https://reviewable.io/reviews/harness/harness-core/30657#-Mw8wXXq9mhdwdBepPR2:-Mw8wXXq9mhdwdBepPR3:b19um5p) ([raw file](https://github.com/harness/harness-core/blob/9506f491e8e30122141540c2ec93cdc76d670d7b/400-rest/src/main/java/io/harness/delegate/resources/DelegateNgTokenResource.java#L80)):*
> ```Java
>   @Timed
>   @ExceptionMetered
>   @AuthRule(permissionType = MANAGE_DELEGATES)
> ```

Are we allowing all to create token by removing the permission here? is that on purpose?

File: 400-rest/src/main/java/io/harness/delegate/resources/DelegateNgTokenInternalResource.java (hunk 2 above)
Comment (arpit3107) [reply 400-rest/src/main/java/io/harness/delegate/resources/DelegateNgTokenInternalResource.java:66]: They are not mandatory. These are for account/org/project level tokens.

**Review by arpit3107 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30657)* status: 7. unresolved discussions. 0 of 1 approvals obtained. 

___
*[120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateTokenNgResource.java, line 50 at r1](https://reviewable.io/reviews/harness/harness-core/30657#-MvXYRdUDebR-KY_yVO1-r1-50:-MwAENMyBR_k6IxGxf3P:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c677f2decef75495a11c27e79da6ff01fb2fc07f/120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateTokenNgResource.java#L50)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

or let it be
</blockquote></details>

Done.
___
*[400-rest/src/main/java/io/harness/delegate/resources/DelegateNgTokenInternalResource.java, line 81 at r2](https://reviewable.io/reviews/harness/harness-core/30657#-Mw7L6ex54FpPnWjwgZk-r2-81:-MwAEFGX9q5ayppIlaQb:b-lr0nq1) ([raw file](https://github.com/harness/harness-core/blob/91cd245c4a84fd0c05275a1ada62e25b5c821142/400-rest/src/main/java/io/harness/delegate/resources/DelegateNgTokenInternalResource.java#L81)):*
<details><summary><i>Previously, gaurav-jain1 (Gaurav Jain) wrote…</i></summary><blockquote>

ditto here and below.
</blockquote></details>

they are different, GET/POST/PUT
___
*[400-rest/src/main/java/io/harness/delegate/resources/DelegateNgTokenInternalResource.java, line 103 at r2](https://reviewable.io/reviews/harness/harness-core/30657#-Mw7L6ex54FpPnWjwgZk-r2-103:-MwAEIv40vm0ublDwCj4:b-eqxlwd) ([raw file](https://github.com/harness/harness-core/blob/91cd245c4a84fd0c05275a1ada62e25b5c821142/400-rest/src/main/java/io/harness/delegate/resources/DelegateNgTokenInternalResource.java#L103)):*
<details><summary><i>Previously, gaurav-jain1 (Gaurav Jain) wrote…</i></summary><blockquote>

If org and projectIds are optional, than will owner creation throw exception ?
</blockquote></details>

Owner will be null in that case, which means its account level token
___
*[400-rest/src/main/java/io/harness/delegate/resources/DelegateNgTokenResource.java, line 80 at r2](https://reviewable.io/reviews/harness/harness-core/30657#-Mw8wXXq9mhdwdBepPR2:-MwAE4Tz0bHXBa5C3BgR:bmfya83) ([raw file](https://github.com/harness/harness-core/blob/9506f491e8e30122141540c2ec93cdc76d670d7b/400-rest/src/main/java/io/harness/delegate/resources/DelegateNgTokenResource.java#L80)):*
<details><summary><i>Previously, jennyjames21 (Jenny James) wrote…</i></summary><blockquote>

Are we allowing all to create token by removing the permission here? is that on purpose?
</blockquote></details>

Access control client will take care of it.



_Code quote (from 120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateTokenNgResource.java):_
```Java
accessControlClient.checkForAccessOrThrow(ResourceScope.of(accountIdentifier, orgIdentifier, projectIdentifier),
```
___
*[400-rest/src/main/java/io/harness/delegate/resources/DelegateTokenResourceForNgManager.java, line 23 at r1](https://reviewable.io/reviews/harness/harness-core/30657#-MvUoqkhkkTqS-9sTr8c-r1-23:-MwAE8V65LLCBSnlQets:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c677f2decef75495a11c27e79da6ff01fb2fc07f/400-rest/src/main/java/io/harness/delegate/resources/DelegateTokenResourceForNgManager.java#L23)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

change file name so that it doesnt havename ng manager
</blockquote></details>

Done.
___
*[400-rest/src/main/java/io/harness/delegate/resources/DelegateTokenResourceForNgManager.java, line 46 at r1](https://reviewable.io/reviews/harness/harness-core/30657#-MvUoqkhkkTqS-9sTr8c-r1-46:-MwAE9Iv3knamLwAL7-Q:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c677f2decef75495a11c27e79da6ff01fb2fc07f/400-rest/src/main/java/io/harness/delegate/resources/DelegateTokenResourceForNgManager.java#L46)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

@InternalApi 
</blockquote></details>

Done.

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 4 of 6 files at r1, 2 of 4 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30657)* status: 4. unresolved discussions.

**Review by bharatgoelharness — APPROVED** [review]
> Reviewed 4 of 6 files at r1, 2 of 4 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30657)* status: 4. unresolved discussions.

**Review by bharatgoelharness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30657)* status: 4. unresolved discussions.

**Review by bharatgoelharness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30657)* status: 4. unresolved discussions.

**Pull request merged**